spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Utils.scala

```diff
@@ -570,4 +570,57 @@ object Utils {
     times: Array[(AbstractModule[_ <: Activity, _ <: Activity, T], Long, Long)]): (Long, Long) = {
       times.map(t => (t._2, t._3)).reduce((a, b) => (a._1 + b._1, a._2 + b._2))
   }
+
+  private[nn] def calc(tensor: Tensor[Float], mask: Int): Array[Float] = {
+    def maskInterval: String = {
+      val start = 0
+      val end = (math.pow(2, tensor.size().length) - 1).toInt
+
+      s"mask should between [$start, $end]"
+    }
+    require(mask.toBinaryString.length <= tensor.size().length, s"$maskInterval")
+
+    // because the shape of tensor is reverse of mask, for example,
+    // if the mask is 11 (bit representation) and the shape is [4, 3, 2, 2].
+    // the corresponding dimension is [4, 3] which will return 12 values.
+    calc(tensor, mask.toBinaryString.reverse, 0).toArray
+  }
+
+  /**
+   * recursively calculate tensor max values based on the mask and index.
+   *
+   * for a [4, 3, 2, 2] tensor and 11 mask, it will generate 4*3=12 max values.
```

Comment:
please refine the explaination, for example, what is 11 mask, it would confuse the readers.